baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/DuplicateArgumentTypes.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.Streams;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.suppliers.Suppliers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Type;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.immutables.value.Value;
+
+/**
+ * Development Practices: Writing good unit tests.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "DuplicateArgumentTypes",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.SUGGESTION,
```

Comment:
BLUF: not useful if people aren't forced to ack it somehow, whether by explicitly suppressing or yelling at us about it.

I don't think this is very useful, because nobody will see it when using `./gradlew idea`, and people who import via IntelliJ will see this as an error thanks to a long-standing intellij bug that I failed to fix correctly...
https://github.com/JetBrains/intellij-community/pull/1285

Either make it a warning and deal with the wrath of the -Werror'ers, or an error.

Reply:
Just to confirm (since I don't have much experience with this) - you can enable this independently for repos, right? Will it default to being off, and be able to be enabled at will? I think we'd be happy to deal with the wrath but I don't want to impose it on everyone unless they opt-in.

Reply:
@dansanduleac I don't think I agree, we should be able to merge this to see how it feels, with opt-in, then followup with an excavator, and then make this mandatory.

Same as with all the other improvements like enum casing etc.